spec/integration/puppet_x/puppetlabs/scheduled_task/v1adapter_spec.rb

```diff
@@ -1,7 +1,7 @@
 #!/usr/bin/env ruby
 require 'spec_helper'
 
-require 'puppet/util/windows/taskscheduler' if Puppet.features.microsoft_windows?
+require_relative '../../../../legacy_taskscheduler' if Puppet.features.microsoft_windows?
```

Comment:
One other thing I should have mentioned earlier.

I think we should do one of two things here:

* Guard the require additionally to only load where `!defined?(Win32::Taskscheduler)`
    * downside is unlikely divergent behavior if someone accidentally modifies the Puppet 4 / Puppet 5 code
    * upside is it prevents a number of duplicate redefinition warnings for the constants
* Include some notes that:
    * we're pulling in deprecated code
    * the code being pulled in will override / reopen classes already defined in the Puppet source / gem for versions < 6

I'd prefer the former in this case. Thoughts?

Reply:
The defensive guard was me thinking that somewhere else code had already called `require 'puppet/util/windows/taskscheduler'` (indirectly through `require 'puppet/util/windows'` for instance). But looking at Puppet, that may not be the case. Worth a quick check to see if the guard is unnecessary.

Reply:
When I run the spec tests locally I _do_ get the warnings about the constants. Going to update with the guard.

Reply:
It looks like rspec has loaded `RSpec::Puppet::Win32::TaskScheduler` at some point before this requires and causes tests to fail - it's also the reason we're seeing the warnings about constants, I think. Investigating...

Reply:
Aha! It's being loaded via `rspec-puppet`:

Reply:
You could look for `Win32::TaskScheduler::COM` instead, which isn't defined in rspec-puppet.

Reply:
Good god, why is rspec-puppet doing that...

Reply:
I don't think it's ever the real thing, at this point - and the updated guard doesn't prevent the reloading of those constants.